src/gfx.cpp

```diff
@@ -1185,6 +1190,44 @@ static void GfxBlitter(const Sprite * const sprite, int x, int y, BlitterMode mo
 	BlitterFactory::GetCurrentBlitter()->Draw(&bp, mode, zoom);
 }
 
+/**
+ * Draws a sprite to a new RGBA buffer (see Colour union) instead of drawing to the screen.
+ * Free the memory after use.
+ *
+ * @param spriteId The sprite to draw.
+ * @return Pixel Buffer
+ */
+const uint32* DrawSpriteToRgbaBuffer(SpriteID spriteId)
```

Comment:
Use `std::unique_ptr` to clearly indicate who owns the return memory. Would require changing MallocT to new[] or adding a custom deleter.